src/useradd.c

```diff
@@ -1894,24 +1894,54 @@ static void usr_update (void)
 static void create_home (void)
 {
 	if (access (user_home, F_OK) != 0) {
+		char path[strlen (user_home) + 2];
+		char *bhome, *cp;
+
+		path[0] = '\0';
+		bhome = strdup (user_home);
+		++bhome;
+
 #ifdef WITH_SELINUX
 		if (set_selinux_file_context (user_home) != 0) {
```

Comment:
One more change - this set_selinux_file_context() must happen after the parent directories have been created, else it will apply to the parent directories we create.

Reply:
Ah yes please, I've somewhat moved on from that code. Given my response latency, perhaps best if you roll your proposed changes in.